core/src/main/scala/tapir/SchemaFor.scala

```diff
@@ -53,4 +54,6 @@ object SchemaFor extends SchemaForMagnoliaDerivation {
   implicit def schemaForPart[T: SchemaFor]: SchemaFor[Part[T]] = new SchemaFor[Part[T]] {
     override def schema: Schema = implicitly[SchemaFor[T]].schema
   }
+
+  implicit def oneOf[E, V](extractor: E => V, stringer: E => String)(mapping: (V, SchemaFor[_])*): Discriminator = macro oneOfMacro[E, V]
```

Comment:
why `implicit`? Would it work if `stringer` had a default of `_.toString`?

Reply:
I removed implicit before you made that comment :) Unfortunately macro applications do not support named and/or default arguments

Reply:
We still can make a implicit typeclass for that

Reply:
Or maybe two overloaded methods, one with 1 parameter the other with 2 params?

Reply:
Yes also thought about that, but for now I don't know how to reuse macro implementation or extract anything from it due to the fact that types are instance specific.

Reply:
Apart from that, I tested such case and it doesn't work either:
```
  def oneOf[E, V](extractor: E => V, asString: V => String)(mapping: (V, SchemaFor[_])*): Discriminator = macro OneOfMacro.oneOfMacro[E, V]
  def oneOf[E, V](extractor: E => V)(mapping: (V, SchemaFor[_])*): Discriminator = macro OneOfMacroWithDefault.oneOfMacro[E, V]
```
So one of the `oneOf` methods would have to be named differently, which may not be so bad. We can have `oneOf(extractor, mapping)` and `oneOfWithCustomToString`.

Or I have better idea. We can life implicitly mapping into other object which would have this default conversion!